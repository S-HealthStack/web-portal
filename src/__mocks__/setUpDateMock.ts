let isEnabled = true;

const originalDateNow = Date.now;

Date.now = jest.fn(() => (isEnabled ? 1680842869025 : originalDateNow()));

export function enableDateNowMock() {
  isEnabled = true;
}

export function disableDateNowMock() {
  isEnabled = false;
}
