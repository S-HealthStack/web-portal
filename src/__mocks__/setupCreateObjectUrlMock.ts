global.URL.createObjectURL = jest.fn(() => 'blob:');
