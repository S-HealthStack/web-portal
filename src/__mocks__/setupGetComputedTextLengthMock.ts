beforeAll(() => {
  (SVGElement.prototype as SVGTextContentElement).getComputedTextLength = () => 0;
});
