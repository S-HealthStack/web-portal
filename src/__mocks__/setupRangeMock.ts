Range.prototype.getBoundingClientRect = () => ({
  bottom: 0,
  height: 0,
  left: 0,
  x: 0,
  y: 0,
  right: 0,
  top: 0,
  width: 0,
  toJSON: jest.fn(),
});

Range.prototype.getClientRects = () => ({
  item: () => null,
  length: 0,
  [Symbol.iterator]: jest.fn(),
});
