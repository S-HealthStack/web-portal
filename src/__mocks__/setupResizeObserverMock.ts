import RO from 'resize-observer-polyfill';

global.ResizeObserver = RO;
