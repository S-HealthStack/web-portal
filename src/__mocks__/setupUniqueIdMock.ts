let counter = 0;
jest.mock(
  'lodash/uniqueId',
  () =>
    (prefix = '') =>
      // eslint-disable-next-line no-plusplus
      `${prefix}${++counter}`
);

beforeEach(() => {
  counter = 0;
});
