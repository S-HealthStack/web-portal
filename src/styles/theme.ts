import { css, DefaultTheme } from 'styled-components';
import { px } from './utils';

const colors = {
  // gray gradations
  black: '#000000',
  black40: 'rgba(0, 0, 0, 0.4)',
  black08: 'rgba(0, 0, 0, 0.08)',

  // light mode
  surface: '#FFFFFF',
  background: '#F7F8FA',
  onPrimary: '#FFFFFF',
  backgroundOnPrimary: '#FFFFFF',
  backgroundSurface: '#FFFFFF',
  backgroundOnSurface: '#2D2D2D',
  primary: '#4475E3',
  primaryLight: '#ECF1FC',
  primaryHovered: '#2F65E0',
  primary05: '#F6F8FE',
  primary10: '#ECF1FC',
  primary30: '#C7D6F7',
  primaryDisabled: '#B3C6F1',
  primaryLightPressed: '#E3EAFB',
  primaryLightFocused: '#C7D6F7',
  textPrimary: '#474747',
  primaryWhite: '#FFFFFF',
  disabled: '#DBDBDB',
  disabled20: 'rgba(219, 219, 219, 0.2)',
  onDisabled: '#A1A1A1',

  textPrimaryBlue: '#4475E3',
  textPrimaryDark: '#474747',
  textSecondaryGray: '#808080',
  textDisabled: '#838383',
  onSurface: '#2D2D2D',
  secondarySkyblue: '#00B0D7', // replace
  backgoundLight: '#F0F0F0',
  primaryBlueHovered: '#2F65E0',
  primaryBluePressed: '#2057D5',

  statusSuccess: '#2DC008',
  statusSuccess10: '#DCF2EA',
  statusSuccessText: '#00825D',

  statusError: '#FF3333',
  statusError10: '#FFEBEB',
  statusErrorText: '#D14343',

  statusWarning: '#F5BF00',
  statusWarning10: '#FFF7D0',
  statusWarningText: '#A36400',

  secondaryViolet: '#944ED7',
  secondaryVioletHovered: '#6A3799',
  secondaryViolet10: '#F4EDFB',

  secondaryTangerine: '#F39C18',
  secondaryTangerineHovered: '#CC8314',
  secondaryTangerine10: '#FEF5E8',

  secondarySkyBlue: '#00B0D7',
  secondarySkyBlueHovered: '#0092B3',
  secondarySkyBlue10: '#E5F7FB',

  secondaryGreen: '#1AD598',
  secondaryGreenHovered: '#16B37F',
  secondaryGreen10: '#E8FBF5',

  secondaryRed: '#FA5F4F',
  secondaryRedHovered: '#CC4E40',
  secondaryRed10: '#FFEFED',
};

export type SpecColorType = keyof typeof colors;

const weightToCssValue = {
  regular: 400,
  upperregular: 450,
  medium: 500,
  semibold: 600,
  bold: 700,
};

const fontSpec = (
  size: number,
  lineHeight: number | string,
  weight: keyof typeof weightToCssValue,
  fontFamily: string
) => css`
  font-family: ${fontFamily}, sans-serif;
  font-size: ${px(size)};
  line-height: ${typeof lineHeight === 'string' ? lineHeight : px(lineHeight)};
  font-weight: ${weightToCssValue[weight]};
  font-style: normal;
`;

const typography = {
  // Inter typeface
  labelRegular: fontSpec(10, 15, 'regular', 'Inter'),
  labelSemibold: fontSpec(10, 15, 'semibold', 'Inter'),

  bodyXSmallRegular: fontSpec(12, 18, 'regular', 'Inter'),
  bodyXSmallSemibold: fontSpec(12, 18, 'semibold', 'Inter'),
  bodySmallRegular: fontSpec(14, 21, 'regular', 'Inter'),
  bodySmallSemibold: fontSpec(14, 21, 'semibold', 'Inter'),
  bodyMediumRegular: fontSpec(16, 24, 'regular', 'Inter'),
  bodyMediumSemibold: fontSpec(16, 24, 'semibold', 'Inter'),
  bodyLargeRegular: fontSpec(18, 26, 'regular', 'Inter'),
  bodyLargeSemibold: fontSpec(18, 26, 'semibold', 'Inter'),

  headingXSmall: fontSpec(16, 21, 'semibold', 'Inter'),
  headingSmall: fontSpec(18, 23, 'semibold', 'Inter'),
  headingXMedium: fontSpec(18, 27, 'semibold', 'Inter'),
  headingXMediumRegular: fontSpec(18, 27, 'regular', 'Inter'),
  headingMedium: fontSpec(24, 32.2, 'semibold', 'Inter'),
  headingLargeSemibold: fontSpec(32, 42, 'semibold', 'Inter'),
  headingXLargeSemibold: fontSpec(48, 58, 'semibold', 'Inter'),
  headingXXLargeSemibold: fontSpec(60, 72, 'semibold', 'Inter'),

  // Fira Code typeface
  query14: fontSpec(14, '130%', 'upperregular', 'Fira Code'),
  query12: fontSpec(12, '150%', 'upperregular', 'Fira Code'),
};

const boxShadow = {
  card: `0 0 ${px(2)} ${px(0)} rgba(0,0,0,0.15)`,
  avatar: `0 ${px(8)} ${px(10)} ${px(2)} 0 rgba(0,0,0,0.12)`,
  studyAvatar: `0 ${px(15)} ${px(20)} rgba(180,180,180,0.3)`,
  previewScreen: `0 ${px(2)} ${px(4)} rgba(71,71,71,0.25)`,
};

const animation = {
  defaultTiming: 'cubic-bezier(0.45, 0.05, 0, 1)',
  rippleTiming: 'cubic-bezier(0.4, 0, 0.2, 1)',
};

declare module 'styled-components' {
  export interface DefaultTheme {
    colors: typeof colors;
    typography: typeof typography;
    boxShadow: typeof boxShadow;
    animation: typeof animation;
  }
}

const theme: DefaultTheme = {
  colors,
  typography,
  boxShadow,
  animation,
};

export default theme;
