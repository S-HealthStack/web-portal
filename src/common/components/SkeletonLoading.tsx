import styled, { keyframes } from 'styled-components';
import { animation, px } from 'src/styles';

// TODO: should depend on element size
const BACKGROUND_SIZE = 400;

const SkeletonAnimation = keyframes`
  0% {
    background-position: 0 0;
  }

  100% {
    background-position: ${px(BACKGROUND_SIZE)} 0;
  }
`;

const SkeletonLoading = styled.div`
  border-radius: ${px(4)};
  background: linear-gradient(112.52deg, #e6e6e6 0%, rgba(217, 217, 217, 0) 50%, #e6e6e6 100%);
  background-size: ${px(BACKGROUND_SIZE)} ${px(100)};
  animation: ${SkeletonAnimation} 1s infinite ${animation.defaultTiming};
`;

export default SkeletonLoading;
