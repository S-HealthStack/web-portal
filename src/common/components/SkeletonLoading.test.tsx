import React from 'react';
import '@testing-library/jest-dom';
import 'jest-styled-components';
import 'src/__mocks__/setupUniqueIdMock';
import { render } from '@testing-library/react';
import theme from 'src/styles/theme';
import { ThemeProvider } from 'styled-components/';
import SkeletonLoading, { SkeletonRect } from './SkeletonLoading';

describe('SkeletonLoading', () => {
  it('test skeleton loading render', () => {
    const { baseElement, getByTestId } = render(
      <ThemeProvider theme={theme}>
        <SkeletonLoading data-testid="skeleton-loading">
          <SkeletonRect x="0" y="0" rx="4" width={180} height={24} data-testid="skeleton-rect" />
        </SkeletonLoading>
      </ThemeProvider>
    );

    expect(baseElement).toMatchSnapshot();

    const loading = getByTestId('skeleton-loading');
    const rect = getByTestId('skeleton-rect');

    expect(loading).toBeInTheDocument();
    expect(rect).toBeInTheDocument();
  });

  it('[NEGATIVE] should render with wrong props', () => {
    const { baseElement, getByTestId } = render(
      <ThemeProvider theme={theme}>
        <SkeletonLoading data-testid="skeleton-loading">
          <SkeletonRect x={-1} y={-1} rx={-1} width={-1} height={-1} data-testid="skeleton-rect" />
        </SkeletonLoading>
      </ThemeProvider>
    );

    expect(baseElement).toMatchSnapshot();

    const loading = getByTestId('skeleton-loading');
    const rect = getByTestId('skeleton-rect');

    expect(loading).toBeInTheDocument();
    expect(rect).toBeInTheDocument();
  });
});
