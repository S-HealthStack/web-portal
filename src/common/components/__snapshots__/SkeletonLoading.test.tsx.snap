// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkeletonLoading [NEGATIVE] should render with wrong props 1`] = `
<body>
  <div>
    <svg
      data-testid="skeleton-loading"
      height="-0.125rem"
      preserveAspectRatio="none"
      version="1.1"
      viewBox="0 0 -2 -2"
      width="-0.125rem"
    >
      <defs>
        <radialgradient
          cx="50%"
          cy="50%"
          fx="50%"
          fy="50%"
          gradientTransform="translate(0, -0.5),skewX(-7),scale(0.5, 2),translate(0, 0)"
          id="grad-1"
          r="50%"
        >
          <stop
            offset="0%"
            stop-color="#F3F3F3"
          />
           
          <stop
            offset="70%"
            stop-color="#ECECEC"
          />
           
          <stop
            offset="100%"
            stop-color="#E5E5E5"
            stop-opacity="0"
          />
           
          <animatetransform
            additive="sum"
            attributeName="gradientTransform"
            dur="1300ms"
            from="-1 0"
            repeatCount="indefinite"
            to="8.5 0"
            type="translate"
          />
        </radialgradient>
        <rect
          data-testid="skeleton-rect"
          fill="#E9E9E9"
          height="-1"
          id="skeleton-rect-1-0"
          rx="-1"
          width="-1"
          x="-1"
          y="-1"
        />
        <rect
          fill="url(#grad-1)"
          height="100%"
          id="overlay-1"
          stroke="none"
          width="100%"
          x="0"
          y="0"
        />
        <clippath
          id="elements-clip-1"
        >
          <use
            href="#skeleton-rect-1-0"
          />
        </clippath>
      </defs>
      <use
        href="#skeleton-rect-1-0"
      />
      <use
        clip-path="url(#elements-clip-1)"
        href="#overlay-1"
      />
    </svg>
  </div>
</body>
`;

exports[`SkeletonLoading test skeleton loading render 1`] = `
<body>
  <div>
    <svg
      data-testid="skeleton-loading"
      height="1.5rem"
      preserveAspectRatio="none"
      version="1.1"
      viewBox="0 0 180 24"
      width="11.25rem"
    >
      <defs>
        <radialgradient
          cx="50%"
          cy="50%"
          fx="50%"
          fy="50%"
          gradientTransform="translate(0, -0.5),skewX(-7),scale(0.5, 2),translate(0, 0)"
          id="grad-1"
          r="50%"
        >
          <stop
            offset="0%"
            stop-color="#F3F3F3"
          />
           
          <stop
            offset="70%"
            stop-color="#ECECEC"
          />
           
          <stop
            offset="100%"
            stop-color="#E5E5E5"
            stop-opacity="0"
          />
           
          <animatetransform
            additive="sum"
            attributeName="gradientTransform"
            dur="1300ms"
            from="-1 0"
            repeatCount="indefinite"
            to="8.5 0"
            type="translate"
          />
        </radialgradient>
        <rect
          data-testid="skeleton-rect"
          fill="#E9E9E9"
          height="24"
          id="skeleton-rect-1-0"
          rx="4"
          width="180"
          x="0"
          y="0"
        />
        <rect
          fill="url(#grad-1)"
          height="100%"
          id="overlay-1"
          stroke="none"
          width="100%"
          x="0"
          y="0"
        />
        <clippath
          id="elements-clip-1"
        >
          <use
            href="#skeleton-rect-1-0"
          />
        </clippath>
      </defs>
      <use
        href="#skeleton-rect-1-0"
      />
      <use
        clip-path="url(#elements-clip-1)"
        href="#overlay-1"
      />
    </svg>
  </div>
</body>
`;
